d4s/src/main/scala/d4s/codecs/D4SAttributeEncoder.scala

```diff
@@ -80,14 +80,7 @@ object D4SAttributeEncoder {
       item.map(T.encodeAttribute).getOrElse(AttributeValue.builder().nul(true).build())
   }
 
-//  implicit def eitherEncoder[A: D4SAttributeEncoder, B: D4SAttributeEncoder](keyLeft: String, keyRight: String): D4SAttributeEncoder[Either[A, B]] = {
-//    item =>
-//      val map = item match {
-//        case Left(value)  => Map(keyLeft  -> D4SAttributeEncoder.encodeAttribute(value))
-//        case Right(value) => Map(keyRight -> D4SAttributeEncoder.encodeAttribute(value))
-//      }
-//      AttributeValue.builder().m(map.asJava).build()
-//  }
+  implicit def eitherEncoder[A: D4SAttributeEncoder, B: D4SAttributeEncoder]: D4SAttributeEncoder[Either[A, B]] = D4SEncoder.nonСastedGen[Either[A, B]]
```

Comment:
Cast it to `D4SEncoder`, otherwise it can't be used where `D4SEncoder` is expected

Reply:
Done ✅

Reply:
@vpodlnk err, not done, the type above is still `: D4SAttributeEncoder[Either[A, B]]`